### DISCUSSION TITLE
Deleting table with 'Links' [May 26]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2246] [Discord Threads]
> 🙋‍♂️ **AK**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### You will be required to delete 'Link to another record' column (relational column) before deleting the table.  
#  
> 🙋‍♂️ **AK**  
#### I want to delete the table's rows (empty it from data), not delete all the table. Maybe I'm using the wrong endpoint?  
#  
> 💁 **o1lab**  
#### you would have to delete the related rows to begin with then delete these rows  
#  
> 💁 **o1lab**  
#### or you would have to set cascade on delete option while creating link to another record  
#  
> 🙋‍♂️ **AK**  
#### I re-create with cascade but it's until not working.  I also tried to delete all the rows in all the tables in case because it created and related to the table before it will make me problem, but until the same error...  
#  
> 🙋‍♂️ **AK**  
#### Any suggestions? 🙏  
#  
> 🙋‍♂️ **AK**  
#### I'm still looking for solution 🙏   
#### Can it be bug or It something I'm doing worng?  
#  
> 👨‍🦲 **dstala**  
#### <@396976051172933635>, share your project information please (`copy project info` from `:` menu on right top).  
#  
> 🙋‍♂️ **AK**  
#### Node: **v12.22.12**   
#### Arch: **x64**   
#### Platform: **linux**   
#### Docker: **false**   
#### Database: **pg**   
#### ProjectOnRootDB: **true**   
#### RootDB: **pg**   
#### PackageVersion: **0.90.11**  
#  
> 🙋‍♂️ **AK**  
#### <@857998597474484295>  
#  
> 👨‍🦲 **dstala**  
#### <@396976051172933635>, apologies for delayed response.   
#### on PG, I was able to clear records with 'CASCADE' on delete option set during LinkToAnotherRecord creation.   
####    
#### ```   
####   await api.dbTableColumn.create(tblCity.id, {   
####     title: "countryRead",   
####     uidt: UITypes.LinkToAnotherRecord,   
####     parentId: tblCity.id,   
####     childId: tblCountry.id,   
####     type: 'mm',   
####     onDelete: 'CASCADE'   
####   });   
####    
####   await api.dbTableRow.bulkCreate("nc", project.id, tblCity.id, [{Title: "a1"}, {Title: "a2"}, {Title: "a3"}, {Title: "a4"}])   
####   await api.dbTableRow.bulkCreate("nc", project.id, tblCountry.id, [{Title: "b1"}, {Title: "b2"}, {Title: "b3"}, {Title: "b4"}])   
####   for (let i=1; i<=4; i++)   
####     await api.dbTableRow.nestedAdd("nc", project.id, tblCity.id, `${i}`, "mm", "countryRead", `${i}`);   
####   await api.dbTableRow.bulkDeleteAll("nc", project.id, tblCity.id, ["1", "2", "3", "4"])   
####   await api.dbTableRow.bulkDeleteAll("nc", project.id, tblCountry.id, ["1", "2", "3", "4"])   
#### ```  
#  
> 🙋‍♂️ **AK**  
#### I will try to re create it from code with the sdk and update if now it is working.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/975087237849038979)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
